unifiedWSHandler.js

```diff
@@ -38,75 +38,68 @@ const environmentToPlatform = {
  * @param {discord.Client} discordClient - discord client to use
  * @throws if there are error in logEvent
  */
-const logEvent = async function(payload, environment, pgClient, discordClient){
-    let response = await censusRequest(environment, 'character_list', `/character/${payload.character_id}?c:resolve=outfit_member`);
-    let platform = environmentToPlatform[environment];
-    let playerEvent = payload.event_name.substring(6);
-    if(response[0].outfit_member != null){
-        let char = response[0];
-        let result = {};
-        try{
-            result = await pgClient.query("SELECT a.id, a.color, a.alias, a.channel, a.platform, c.autoDelete\
+async function logEvent(payload, environment, pgClient, discordClient){
+    const response = await censusRequest(environment, 'character_list', `/character/${payload.character_id}?c:resolve=outfit_member`);
+    if(response[0].outfit_member === undefined){
+        return;
+    }
+    const char = response[0];
+    let result = {};
+    try{
+        const platform = environmentToPlatform[environment];
+        result = await pgClient.query("SELECT a.id, a.color, a.alias, a.channel, a.platform, c.autoDelete\
             FROM outfitActivity a LEFT JOIN subscriptionConfig c ON a.channel = c.channel\
             WHERE a.id = $1 AND a.platform = $2;", [char.outfit_member.outfit_id, platform]);
+        if (result.rows.length === 0){
+            return
         }
-        catch(error){
-            throw `Error in logEvent: ${error}`;
+    }
+    catch(error){
+        throw `Error in logEvent: ${error}`;
+    }
+    const sendEmbed = new MessageEmbed();
+    const playerEvent = payload.event_name.substring(6);
+    sendEmbed.setTitle(`${result.rows[0].alias} ${playerEvent}`);
+    sendEmbed.setDescription(char.name.first);
+    sendEmbed.setColor(faction(char.faction_id).color);
+    Promise.allSettled(result.rows.map(async row => {
+        try {
+            const resChann = await discordClient.channels.fetch(row.channel);
+            if (resChann.type === 'DM'){ // DM
+                const messageId = await messageHandler.send(resChann, {embeds: [sendEmbed]}, "Log event");
+                if(messageId != -1 && row.autodelete === true){
+                    const in5minutes = new Date((new Date()).getTime() + 30000);
+                    pgClient.query("INSERT INTO toDelete (channel, messageID, timeToDelete) VALUES ($1, $2, $3)", [row.channel, messageId, in5minutes])
+                        .catch(err => {console.log(err);});
+                }
+                else if(messageId == -1){
```

Comment:
How can we check if a user has blocked the bot or closed dms?  I'd feel more comfortable with that being the metric instead of a failed message to unsubscribe in DMs.

Reply:
Unfortunately it seems like there is  no way to check is a user has the bot blocked or has closed DMs. Discord will throw an error saying the message couldn't be sent.